### PR TITLE
[MIR-trans] Small fixes for the initial trans support

### PR DESCRIPTION
**Pull request by dotdash**
Fix handling of small aggregate function arguments and assignments of temporaries to lvalues.

### CODE REVIEW

File: src/librustc_trans/trans/mir/operand.rs [hunk 1]
```diff
@@ -94,6 +94,24 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
 
         match *operand {
             mir::Operand::Consume(ref lvalue) => {
+                // watch out for temporaries that do not have an
```
Comment (nikomatsakis) [inline src/librustc_trans/trans/mir/operand.rs]: I wonder if this "trans operand into" path is all that useful. It seems like we could generate the same code by just calling `trans_operand` and then store the resulting operand instead, so as not to duplicate this logic.

Comment (dotdash) [inline src/librustc_trans/trans/mir/operand.rs]: Oh right! I wanted to preserve the memcpy for lvalue-to-lvalue copies, but for immediate values, the load/store combo is even actually better I suppose.

**Comment by nikomatsakis** [comment]
@bors r+

nice

**Comment by dotdash** [comment]
Added one more commit to fix handling of fat pointer function arguments.

**Pull request merged**